crates/biome_js_analyze/src/semantic_analyzers/nursery/no_invalid_new_builtin.rs

```diff
@@ -0,0 +1,107 @@
+use crate::{semantic_services::Semantic, JsRuleAction};
+use biome_analyze::{context::RuleContext, declare_rule, ActionCategory, Rule, RuleDiagnostic};
+use biome_console::markup;
+use biome_diagnostics::Applicability;
+use biome_js_factory::make;
+use biome_js_syntax::{
+    global_identifier, static_value::StaticValue, AnyJsExpression, JsCallExpression,
+    JsNewExpression,
+};
+use biome_rowan::{chain_trivia_pieces, AstNode, BatchMutationExt};
+
+declare_rule! {
+    /// Disallow `new` operators with global non-constructor functions.
+    ///
+    /// Some global functions cannot be called using the new operator and
+    /// will throw a `TypeError` if you attempt to do so. These functions are:
+    ///
+    /// - [`Symbol`](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Symbol/Symbol)
+    /// - [`BigInt`](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/BigInt/BigInt)
+    ///
+    /// Source: https://eslint.org/docs/latest/rules/no-new-native-nonconstructor/
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// var foo = new Symbol('foo');
+    /// var bar = new BigInt(9007199254740991);
+    /// ```
+    ///
+    /// ## Valid
+    ///
+    /// ```js
+    /// var foo = Symbol('foo');
+    /// var bar = BigInt(9007199254740991);
+    ///
+    /// function baz(Symbol) {
+    ///     const qux = new Symbol("baz");
+    /// }
+    /// function quux(BigInt) {
+    ///     const corge = new BigInt(9007199254740991);
+    /// }
+    /// ```
+    pub(crate) NoInvalidNewBuiltin {
+        version: "next",
+        name: "noInvalidNewBuiltin",
+        recommended: true,
+    }
+}
+
+impl Rule for NoInvalidNewBuiltin {
+    type Query = Semantic<JsNewExpression>;
+    type State = StaticValue;
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let callee = ctx.query().callee().ok()?;
+        let (reference, name) = global_identifier(&callee)?;
+        match name.text() {
+            "Symbol" | "BigInt" => ctx.model().binding(&reference).is_none().then_some(name),
```

Comment:
https://github.com/biomejs/biome/pull/375/commits/38964a6496f720d5fce96262d9c9d0b7adb5b45d also handles `Intl`.

Reply:
@Conaclos what are your thoughts on https://github.com/biomejs/biome/pull/375#issuecomment-1731144645? Should we handle the global objects in this rule, or leave them to `noGlobalObjectCalls`?

Reply:
Ok I completely forgot about this other rule :/

I am really sorry! I think we should undo my suggestion because [noGlobalObjectCalls](https://biomejs.dev/linter/rules/no-global-object-calls/) already handle thes ecases.

Reply:
No worries! I can revert https://github.com/biomejs/biome/pull/375/commits/38964a6496f720d5fce96262d9c9d0b7adb5b45d :)